### ISSUE TITLE
Attempting to create an evaluation screenshot crashes with Unknown pixel format

### ISSUE THREAD

**Issue by ghost** [stepmania/stepmania#1957]
Always reproducible crash when pressing P1Select on the song evaluation screen to create a screenshot to memory card. Also happens on latest commit. Not sure what has changed meanwhile. No system changes have been done since. According to my profile, the last successful screenshot was on 2019-12-23.

[crashinfo.txt](https://github.com/stepmania/stepmania/files/4148295/crashinfo.txt)

```
StepMania5.1-git-49b99e702c
Compiled 20191231 @ 14:06:47 (build 49b99e702c)
Log starting 2020-02-03 13:45:13
Loading window: win32
Windows 6.1 (Win7) build 7601 [Service Pack 1]
Memory: 2047mb total, 3032mb swap (2334mb swap avail)
Video driver: Intel(R) G45/G43 Express Chipset [Intel Corporation]
              8.15.10.2021, 12-14-2009 [pci\ven_8086&dev_2e22]
```
```
Crash reason: Unknown pixel format

Thread: Main thread
        RageTextureManager.cpp:144 RageTextureManager::LoadTexture(/Themes/Simply-Love-sm5-znewver/Graphics/_VisualStyles/Arrows/SharedBackground.png).
        RageTextureManager.cpp:144 RageTextureManager::LoadTexture(/Themes/Simply-Love-sm5-znewver/Graphics/_VisualStyles/Arrows/SharedBackground.png).
        GameSoundManager.cpp:583 126.994080, delta 0.734000
        Screen.cpp:164 ScreenMessage(SM_GainFocus)
        RageDisplay_D3D.cpp:677 Unknown pixel format
```

**Comment by coderjo** [comment]
What video mode are you running (mainly, what color depth: 16 bit, 24 bit, 32 bit)?

(I really should have made that message output the pixel format, but I wasn't really expecting anyone would hit it.)

**Comment by ghost** [comment]
16 bit. Changing to 32 bit makes it work again. I don't recall ever changing this, but I'm just gonna keep it on 32 if there's no performance downside.

**Comment by coderjo** [comment]
A while back, I removed the requirement on the directx SDK, but that meant I had to rewrite the screenshot code because it depended on a helper library that wasn't in the windows SDK. I didn't think anyone was still using 16 bit, so I left it implemented.
I don't remember offhand when I committed that, but if you recently upgraded, your old version may have been before that change.

**Comment by quietly-turning** [comment]
>  I didn't think anyone was still using 16 bit, so I left it implemented.

I'm out of the loop here, but I'm assuming we're talking about the `TextureColorDepth` preference?  

I see people sharing photos of scores where their `TextureColorDepth` is almost certainly set to 16bit pretty regularly.  I've always assumed this is because most people don't know to go into the Graphics/Sound Options menu and/or don't know what the *Texture Color* setting does.

I was going to suggest changing the default value to be `32`, but it looks like that was already done way back in 9e8b8c3c09685f85efe8950cce20556d75e884ee.  Testing it out right now on a macOS build of 5_1-new at 500b03a611888cca57ab0e50c5e7171bf7613d04, it looks like `TextureColorDepth` is set to `16` by default when a new Preferences.ini file is created.

EDIT: Wait, no, you're probably talking about `DisplayColorDepth`.

**Comment by coderjo** [comment]
No, not TextureColorDepth, DisplayColorDepth, aka the color depth the display buffer is in, since we have to create an identical buffer to copy it to.

The helper library (D3DX) had a helper function that would convert the buffer to the desired color depth easily. Without that, I need to copy from the back buffer to a temporary buffer, then copy that to a 32 bit buffer, since that is what the code that is requesting the screenshot expects.

(see [this comment](https://github.com/stepmania/stepmania/blob/5_1-new/src/RageDisplay_D3D.cpp#L669) )

**Comment by coderjo** [comment]
Basically, since I didn't think anyone still was using 16 bit video modes, I took the easier and faster route. But I was wrong.
